mbzirc_ign/models/waves/materials/programs/GerstnerWaves_vs_330.glsl

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2016 The UUV Simulator Authors.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.s
+
+
+// Copyright (c) 2019 Rhys Mainwaring.
+//
+// Modified to accept vector parameters and use the form
+// for Gerstner waves published in:
+//
+// Jerry Tessendorf, "Simulating Ocean Water", 1999-2004
+//
+// theta = k * dir . x - omega * t
+//
+// px = x - dir.x * a * k * sin(theta)
+// py = y - dir.y * a * k * sin(theta)
+// pz =         a * k * cos(theta)
+//
+// k is the wavenumber
+// omega is the angular frequency
+//
+// The derivative terms (Tangent, Binormal, Normal) have been
+// updated to be consistent with this convention.
+//
+
+#version 330
+
+in vec4 vertex;
+in vec4 uv0;
+uniform mat4 worldviewproj_matrix;
+
+/////////// Input parameters //////////
+// Waves
+uniform int Nwaves;
+uniform vec3 camera_position_object_space;
```

Comment:
nit: camelCase?

Reply:
same as previous comment about mapping to predefined constant.